src/Aspire.Hosting.Azure/AzureCosmosDBResource.cs

```diff
@@ -22,5 +22,32 @@ public class AzureCosmosDBResource(string name, string? connectionString)
     /// Gets the connection string to use for this database.
     /// </summary>
     /// <returns>The connection string to use for this database.</returns>
-    public string? GetConnectionString() => ConnectionString;
+    public string? GetConnectionString() => IsEmulator
+        ? AzureCosmosDBEmulatorConnectionString.Create(GetEmulatorPort("emulator"))
+        : ConnectionString;
+
+    /// <summary>
+    /// Gets a value indicating whether the Azure Cosmos DB resource is running in the local emulator.
+    /// </summary>
+    public bool IsEmulator => this.IsContainer();
+
+    private int GetEmulatorPort(string endpointName) =>
+        Annotations
+            .OfType<EndpointAnnotation>()
```

Comment:
This should be using AllocatedEndpoint not endpoint

Reply:
Why is that the correct one to use? What's the difference between the two, and how would a developer know that the allocated endpoint is the correct one to use? When is it allocated?

Reply:
Also, does that mean that the I'd need to change the call to `WithAnnotation` to pass an `AllocatedEndpointAnnotation` instead?

Reply:
No, but it is confusing. The EndpointAnnotation is the requested endpoint information and allocatedendpoint is when the service (proxy) is assigned a port.

Reply:
Right, when is the proxy port assigned? After `builder.Build` or `app.Run`?

Reply:
During app.Run, not after Build.